### PR TITLE
Implement pathToId and add tests for it

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
For global storage and file storage service

Refs: https://github.com/metarhia/common/issues/147

### CODE REVIEW

File: lib/id.js [hunk 1]
```diff
@@ -137,6 +137,19 @@ const idToPath = id => {
...
+const pathToId = path => {
+  const chunks = path.split('/');
+  const count = chunks.length;
+  let hex = '0x';
+  let i;
+
+  for (i = count - 1; i >= 0; i--) {
+    hex += chunks[i];
+  }
+  const id = parseInt(hex);
```
Comment (tshemsedinov) [inline lib/id.js]: I'd rather use `parseInt(hex, 16)`

File: lib/id.js [hunk 2]
```diff
@@ -137,6 +137,19 @@ const idToPath = id => {
   return path;
 };
 
+const pathToId = path => {
+  const chunks = path.split('/');
+  const count = chunks.length;
+  let hex = '0x';
+  let i;
+
```
Comment (tshemsedinov) [inline lib/id.js]: Empty line is not needed here

File: lib/id.js [hunk 3]
```diff
@@ -137,6 +137,19 @@ const idToPath = id => {
...
+  for (i = count - 1; i >= 0; i--) {
+    hex += chunks[i];
+  }
+  const id = parseInt(hex);
+  return id;
```
Comment (lundibundi) [inline lib/id.js]: Why put id to a variable and not just `return parseInt(hex, 16)`?

File: test/id.js [hunk 4]
```diff
@@ -44,6 +44,14 @@ api.metatests.case('Common / id', {
     [123456789,        'cd15/075b'],
     [123456789123,     '1a83/be99/001c'],
     [9007199254740991, 'ffff/ffff/ffff/001f']
+  ],
+  'common.pathToId': [
+    ['0000/0000',           0],
+    ['0001/0000',           1],
```
Comment (lundibundi) [inline test/id.js]: Add a few corner cases that make sure that id will be different for '0001/0000' and '0000/0001' and such.

File: lib/id.js [hunk 5]
```diff
@@ -137,6 +137,19 @@ const idToPath = id => {
   return path;
 };
 
+const pathToId = path => {
+  const chunks = path.split('/');
+  const count = chunks.length;
+  let hex = '0x';
+  let i;
+
+  for (i = count - 1; i >= 0; i--) {
```
Comment (lundibundi) [inline lib/id.js]: I don't think it's worth to cache length as you only use it once here.

**Comment by tshemsedinov** [comment]
Next time use `commit --fixup` please for such a changes so it will autosquash on landing

**Comment by tshemsedinov** [comment]
One more hint: you can create branch in metarhia not separate repo fork in your account

File: lib/id.js [hunk 6]
```diff
@@ -137,6 +137,16 @@ const idToPath = id => {
   return path;
 };
 
+const pathToId = path => {
+  const chunks = path.split('/');
+  let hex = '0x';
```
Comment (belochub) [inline lib/id.js]: You don't need to add `0x` at the start of the number string if you are using `parseInt()` providing the base to it.
Also, you can make this function a one-liner that way.

Comment (tshemsedinov) [reply lib/id.js]: Sorry, I already landed these PR. But yes, `split.reverse.join` may turn this func to one-lien. In spite of this such an implementation is good enough and may even be faster. @belochub

**Comment by tshemsedinov** [comment]
Landed in f77f5dc

**Comment by tshemsedinov** [comment]
@SemenchenkoVitaliy if you closes issue by commit, you should add `Closes: URL` tag, not `Refs: URL` so github will automatically close that issue.

**Comment by SemenchenkoVitaliy** [comment]
Ok, understood

File: lib/id.js (hunk 6 above)
Comment (belochub) [reply lib/id.js]: @tshemsedinov, you've forgotten to land both commits or to squash commits before landing.

Comment (belochub) [reply lib/id.js]: Also, you can make it a one liner that way:
```js
parseInt(path.split('/').reduceRight((hex, chunk) => hex + chunk), 16);
```

Comment (tshemsedinov) [reply lib/id.js]: @belochub thanks, I'll add it as a separate commit
